crates/proof-of-sql/src/evm_compatibility/dyn_proof_plan_serializer.rs

```diff
@@ -0,0 +1,314 @@
+use crate::{
+    base::{
+        database::{ColumnRef, LiteralValue, TableRef},
+        map::{IndexMap, IndexSet},
+        scalar::Scalar,
+    },
+    sql::{
+        parse::QueryExpr,
+        proof::ProofPlan,
+        proof_exprs::{
+            AliasedDynProofExpr, ColumnExpr, DynProofExpr, EqualsExpr, LiteralExpr, TableExpr,
+        },
+        proof_plans::{DynProofPlan, FilterExec},
+    },
+};
+use core::{fmt::Debug, marker::PhantomData};
+use snafu::{OptionExt, Snafu};
+use std::convert::TryFrom;
+use zerocopy::AsBytes;
+
+const FILTER_EXEC_NUM: u8 = 0;
+const COLUMN_EXPR_NUM: u8 = 0;
+const EQUALS_EXPR_NUM: u8 = 1;
+const LITERAL_EXPR_NUM: u8 = 2;
+const BIGINT_TYPE_NUM: u8 = 0;
+
+#[derive(Debug, Snafu)]
+/// Errors that can occur during serialization
+pub enum SerializationError {
+    #[snafu(display("Not supported"))]
+    /// A SQL expression that is not supported by the serialization
+    NotSupported,
+    #[snafu(display("Too many results"))]
+    /// The number of results in a filter expression is too large
+    TooManyResults,
+    #[snafu(display("Too many tables"))]
+    /// The number of tables in a plan is too large
+    TooManyTables,
```

Comment:
I think we should put the exact limit in the error message or people won't even know how many are too many